addons/website/static/src/systray_items/publish.js

```diff
@@ -61,14 +61,16 @@ class PublishSystray extends Component {
             async (published) => {
                 this.state.published = published;
                 if (published) {
-                    const seo_data = await rpc("/website/get_seo_data", {
-                        res_id: mainObject.id,
-                        res_model: mainObject.model,
-                    });
-                    checkAndNotifySEO(seo_data, OptimizeSEODialog, {
-                        notification: this.notificationService,
-                        dialog: this.dialogService,
-                    });
+                    if (this.website.currentWebsite.metadata.canOptimizeSeo) {
```

Comment:
Can't you incorporate both checks inside their preceding ifs to save some code?

Reply:
@awa-odoo
my bad, take a look now!